docs/guides/syntax-notes.md

```diff
@@ -3,11 +3,11 @@ id: syntax-notes
 title: 'Syntax Notes'
 ---
 
-This Page will show different possibilities of how things can be written in typegoose, with their use-case and what differs between them
+This Page shows different possibilities of how code can be written in Typegoose, along with their use-cases and what differs between them.
 
 ## `type` & `ref` with function or without
 
-The options [`type`](../api/decorators/prop.md#type) and [`ref`](../api/decorators/prop.md#ref) can be written either `type: Type` or `type: () => Type`, both are valid options, but the second should always be preferred when not using Primitives, because the "deferred function" way works-around the issues of [Circular References](./advanced/reference-other-classes.md#circular-dependencies) (in some cases) and use-before-definition cases.
+The options [`type`](../api/decorators/prop.md#type) and [`ref`](../api/decorators/prop.md#ref) can be written either as `type: Type` or as `type: () => Type`. Both syntax variations are valid options, but the second should always be preferred when not using primitives, because this "deferred function" syntax can  workaround the issues of [Circular References](./advanced/reference-other-classes.md#circular-dependencies) (in some cases) and also correct situations, where you might run into `use-before-define` errors.
```

Comment:
```suggestion
The options [`type`](../api/decorators/prop.md#type) and [`ref`](../api/decorators/prop.md#ref) can be written either as `type: Type` or as `type: () => Type`. Both syntax variations are valid options, but the second should always be preferred when not using primitives, because this "deferred function" syntax can  workaround the issues of [Circular References](./advanced/reference-other-classes.md#circular-dependencies) (in some cases) and also correct situations, where you might run into `use-before-definition` errors.
```

Reply:
The ts-lint error is `use-before-define`. Is that not what is meant?

Scott

Reply:
i had it taken from what i recall from the typescript error, but it seems i was slightly wrong:
![error](https://user-images.githubusercontent.com/10911626/222731327-6d8fe213-8d3a-454d-b989-91ea62a72ecf.png)

Reply:
Hmm... totally different. Ok. So, what should we write?

`ERROR: used before its declaration`? 

Scott

Reply:
> ERROR: used before its declaration?

sure